neuromation/api/quota.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from yarl import URL
+
+from neuromation.api.config import _Config
+from neuromation.api.core import _Core
+from neuromation.api.utils import NoPublicConstructor
+
+
+@dataclass(frozen=True)
+class QuotaDetails:
+    spent_minutes: int
```

Comment:
Don't use minutes in API but float seconds.
float("inf") is for infinity, please use this constant instead of `None`

Reply:
> Don't use minutes in API but float seconds.

for this, we will need to change the server-side API. Note, currently, we get the time in minutes (see https://github.com/neuromation/platform-api/issues/901).

Should we change the API?

Reply:
Not necessary, you can multiply by 60 on the client.
In python there are two time representations: float and datetime.
It is confusing enough right now, adding yet another notation makes everything even worse and less usable: people will always convert minutes into somethind more convinient.

Reply:
Why do we need `QuotaDetails`? Why not use just `float` or `Optional[timedelta]`?

Reply:
> Not necessary, you can multiply by 60 on the client.

OK, I will change `..._minutes: int` to `..._seconds: float` everywhere. But for me personally it'd be confusing if I see `GPU spent: 00h 01m 00s`, then I run a job for 30 seconds and I still see: `GPU spent: 00h 01m 00s`. I think, if we need to show time in seconds, the API protocol should be changed.

Reply:
> Why do we need QuotaDetails? Why not use just float or Optional[timedelta]?

In the first version of my code, there were just fields `gpu_spent_minutes`, `cpu_limit_minutes`, etc. But since the lines for `CPU` and `GPU` should use the same formatting method `_format_quota_details`, I found it useful to introduce an additional abstraction `QuotaDetails`. Though, if it's an eyesore for you, I'm OK to get rid of it. Should I? cc @serhiy-storchaka

Reply:
API uses seconds (omit the suffix from field names please).
CLI can stay with minutes, it makes sense.

Like we return file size as bytes in the storage API but show it as gigabytes for large files in CLI.

Reply:
Flat structure with short field names is better IMHO

Reply:
> API uses seconds (omit the suffix from field names please). CLI can stay with minutes, it makes sense.

Fixed.

> Why do we need QuotaDetails? Why not use just float or Optional[timedelta]?

fixed to use a flat structure.